app/code/Magento/InventoryCatalog/Model/DeleteSourceItemsBySkus.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryCatalog\Model;
+
+use Magento\Inventory\Model\ResourceModel\SourceItem\DeleteMultiple;
+use Magento\InventoryApi\Api\GetSourceItemsBySkuInterface;
+use Magento\InventoryCatalogApi\Model\DeleteSourceItemsBySkusInterface;
+
+/**
+ * @inheritDoc
+ */
+class DeleteSourceItemsBySkus implements DeleteSourceItemsBySkusInterface
+{
+    /**
+     * @var GetSourceItemsBySkuInterface
+     */
+    private $getSourceItemsBySku;
+
+    /**
+     * @var DeleteMultiple
+     */
+    private $sourceItemsDelete;
+
+    /**
+     * @param GetSourceItemsBySkuInterface $getSourceItemsBySku
+     * @param DeleteMultiple $sourceItemsDelete
+     */
+    public function __construct(
+        GetSourceItemsBySkuInterface $getSourceItemsBySku,
+        DeleteMultiple $sourceItemsDelete
+    ) {
+        $this->getSourceItemsBySku = $getSourceItemsBySku;
+        $this->sourceItemsDelete = $sourceItemsDelete;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(array $skus): void
```

Comment:
For bulk operations, we should use an approach similar to https://github.com/magento/inventory/pull/2728. 
In this case, we should introduce result object and handle errors for each sku in the array.

Reply:
@sidolov, done